src/pint/models/binary_piecewise.py

```diff
@@ -0,0 +1,465 @@
+            else:
+                raise ValueError(
+                    f"No name or lower bound provided to create a group with"
+                )
+
+        for XR2n in XR2_mapping.values():
+            XR2s[XR2n] = getattr(self, XR2n).quantity
+
+        for xr2_name, xr2_value in XR2s.items():
+            if (xr2_value is not None) and (xr2_name is not None):
+                self.binary_instance.add_binary_params(xr2_name, xr2_value)
+            else:
+                raise ValueError(
+                    f"No name or lower bound provided to create a group with"
+                )
+
+        self.update_binary_object(None)
+
+    def validate(self):
+        """Include catches for overlapping groups. etc
+        Raises
+        ------
+        ValueError
+                if there are pieces with no associated boundaries (T0X_0000 does not have a corresponding PLB_0000/PUB_0000)
+        ValueError
+                if any boundaries overlap (as it makes TOA assignment to a single group ambiguous). i.e. PLB_0000<PLB_0000 and PUB_0000>PLB_0001
+        ValueError
+                if the number of lower and upper bounds don't match (should only be raised if PLB is defined without PUB and validate is run or vice versa)
+        """
+        super().validate()
+        for p in ("T0", "A1"):
+            if getattr(self, p).value is None:
+                raise MissingParameter("BT", p, "%s is required for BT" % p)
+
+        # If any *DOT is set, we need T0
+        for p in ("PBDOT", "OMDOT", "EDOT", "A1DOT"):
+            if getattr(self, p).value is None:
+                getattr(self, p).set("0")
+                getattr(self, p).frozen = True
+
+            if getattr(self, p).value is not None:
+                if self.T0.value is None:
+                    raise MissingParameter("BT", "T0", "T0 is required if *DOT is set")
+
+        if self.GAMMA.value is None:
+            self.GAMMA.set("0")
+            self.GAMMA.frozen = True
+
+        dct_plb = self.get_prefix_mapping_component("PLB_")
+        dct_pub = self.get_prefix_mapping_component("PUB_")
+        dct_T0X = self.get_prefix_mapping_component("T0X_")
+        dct_A1X = self.get_prefix_mapping_component("A1X_")
+        if len(dct_plb) > 0 and len(dct_pub) > 0:
+            ls_plb = list(dct_plb.items())
+            ls_pub = list(dct_pub.items())
+            ls_T0X = list(dct_T0X.items())
+            ls_A1X = list(dct_A1X.items())
+
+            j_plb = [((tup[1]).split("_"))[1] for tup in ls_plb]
+            j_pub = [((tup[1]).split("_"))[1] for tup in ls_pub]
+            j_T0X = [((tup[1]).split("_"))[1] for tup in ls_T0X]
+            j_A1X = [((tup[1]).split("_"))[1] for tup in ls_A1X]
+
+            if j_plb != j_pub:
+                raise ValueError(
+                    f"Group boundary mismatch error. Number of detected upper bounds: {j_plb}. Number of detected upper bounds:{j_pub}"
```

Comment:
probably include a space before the values

Reply:
Added extra space